tests/datasets/test_geo.py

```diff
@@ -483,6 +496,26 @@ def test_change_res(self) -> None:
         assert ds.res == (10.0, 10.0)
         ds.res = 20.0
 
+    def test_nodata_value_is_passed_on(self) -> None:
```

Comment:
I don't fully understand this test, the mocking is too deep. What are you actually trying to test here?

Reply:
The other option ofc is to check that the resulting footprint is correct, meaning the nodata was passed and used correctly. https://github.com/torchgeo/torchgeo/pull/2991 will make that much easier, and I can get rid of the patching.